backend/src/judge/middleware/java.ts

```diff
@@ -56,22 +56,32 @@ export class JavaMiddleware extends JudgeMiddleware {
       );
     }
 
+    let expectedOutput = "";
+    let isEqual = "";
+    if (codePrototype.returnType.includes("[]")) {
+      const output = this.output.replace(/^\[/, "{").replace(/\]$/, "}");
+      expectedOutput = `${codePrototype.returnType} expectedOutput = ${output};`;
+      isEqual = `boolean isEqual = Arrays.equals(res, expectedOutput);`;
+    } else {
+      expectedOutput = `${codePrototype.returnType} expectedOutput = ${this.output};`;
+      isEqual =
+        "boolean isEqual = res.toString().equals(expectedOuput.toString());";
+    }
```

Comment:
Actually can we just stringify every time?

Reply:
Arrays in java need to be handled separately as `arr.toString()` returns hexadecimal of the hashcode which is different for arrays that are equal.